### PR TITLE
feat: ✨ add interactive teaser sample

### LINKED ISSUE #503: feat: ✨ add interactive teaser sample to `sd-teaser`
## User Story
As a user of the Solid Design System, I would like to see a sample how to build interactive `sd-teaser`s that
1. are clickable as a whole to e. g. open a new window
2. change the image "zoom" (has to be clarified by @EllaELSA :)) on hover
3. make two inside the main slot buttons clickable and focusable separately
4. doesn't break valid HTML  (links inside links are not allowed :))

Image

(Hint: You might need something like an absolute button going over the whole element.)

The goal is to easily see and be able to copy-paste a sample with e.g. clickable image in addition to a link and/or a button in the content slot.

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by saemik94**
Closes #503

### CODE REVIEW

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 1]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+    return html`
+      <style>
+        #teaserWithContentPlaceholder::part(media) {
+          flex-grow: 1;
+        }
+      </style>
+      <div class="flex justify-between gap-8">
+        <sd-teaser variant="primary" breakpoint="9999" inset class="flex-1">
+          <div slot="media" class="relative">
+            <img class="aspect-video object-cover" src="./placeholders/generic.jpg" alt="A generic placeholder jpg" />
```
Comment (yoezlem) [inline packages/components/src/components/teaser/teaser.stories.ts:419]: To ensure full accessibility for the image, you would need to describe what is depicted in the image. For example, A **generic placeholder image where a father and son are playing football.**

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 2]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+            </p>
+            <div class="flex-none">
+              <sd-button href="#" target="_blank" variant="primary" inverted>A SD-button</sd-button>
+              <sd-button href="#" target="_blank" variant="primary" inverted>A SD-button</sd-button>
+            </div>
+          </div>
+        </sd-teaser>
+        <sd-teaser variant="white border-neutral-400" breakpoint="9999" inset class="flex-1">
+          <div slot="media" class="relative">
+            <img class="aspect-video object-cover" src="./placeholders/generic.jpg" alt="A generic placeholder jpg" />
```
Comment (yoezlem) [inline packages/components/src/components/teaser/teaser.stories.ts:439]: Also here:
https://github.com/solid-design-system/solid/pull/670/files#r1444476775

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 3]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+            transition-duration: 300ms;
+            opacity: 0.5;
+          }
+        }
+      </style>
+      <div class="flex justify-between gap-8">
+        <a href="#" target="_blank" class="flex-1">
+          <sd-teaser variant="primary" breakpoint="9999" inset class="interactive">
+            <div slot="media" class="relative">
+              <img class="aspect-video object-cover" src="./placeholders/generic.jpg" alt="A generic placeholder jpg" />
```
Comment (yoezlem) [inline packages/components/src/components/teaser/teaser.stories.ts:523]: https://github.com/solid-design-system/solid/pull/670/files#r1444476775 :)

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 4]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+                <sd-button variant="primary" inverted>More</sd-button>
+              </div>
+            </div>
+          </sd-teaser>
+        </a>
+
+        <a href="#" target="_blank" class="flex-1">
+          <sd-teaser variant="white border-neutral-400" breakpoint="9999" inset class="interactive">
+            <div slot="media" class="relative">
+              <img class="aspect-video object-cover" src="./placeholders/generic.jpg" alt="A generic placeholder jpg" />
```
Comment (yoezlem) [inline packages/components/src/components/teaser/teaser.stories.ts:545]: https://github.com/solid-design-system/solid/pull/670/files#r1444476775

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 5]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+          breakpoint="9999"
+          inset
+          class="flex-1 interactive cursor-pointer"
+          data-href="#"
+          data-target="_blank"
+          tabindex="0"
+          role="link"
+        >
+          <div slot="media" class="relative">
+            <img class="aspect-video object-cover" src="./placeholders/generic.jpg" alt="A generic placeholder jpg" />
```
Comment (yoezlem) [inline packages/components/src/components/teaser/teaser.stories.ts:662]: https://github.com/solid-design-system/solid/pull/670/files#r1444476775

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 6]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
```
Comment (yoezlem) [inline packages/components/src/components/teaser/teaser.stories.ts:695]: https://github.com/solid-design-system/solid/pull/670/files#r1444476775

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 7]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+              >
+            </div>
+          </div>
+        </sd-teaser>
+
+        <sd-teaser
+          variant="white border-neutral-400"
+          breakpoint="9999"
+          inset
+          class="flex-1 interactive cursor-pointer"
```
Comment (yoezlem) [inline packages/components/src/components/teaser/teaser.stories.ts:688]:
```suggestion
          class="flex-1 interactive hover:cursor-pointer"
```

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 8]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+            opacity: 0.5;
+          }
+        }
+      </style>
+      <div class="flex justify-between gap-8">
+        <sd-teaser
+          variant="primary"
+          breakpoint="9999"
+          inset
+          class="flex-1 interactive cursor-pointer"
```
Comment (yoezlem) [inline packages/components/src/components/teaser/teaser.stories.ts:655]:
```suggestion
          class="flex-1 interactive hover:cursor-pointer"
```

**Comment by coraliefeil** [comment]
Review done. Pls reassign again when adjusted. THX ;)

File: packages/components/src/components/teaser/teaser.stories.ts (hunk 1 above)
Comment (saemik94) [reply packages/components/src/components/teaser/teaser.stories.ts:419]: In this case it's really just a 'A generic placeholder jpg', or should I use another placeholder image?
image

Comment (yoezlem) [reply packages/components/src/components/teaser/teaser.stories.ts:419]: Ah, okay! No than its not necessary.

**Review by azraefendic — APPROVED** [review]
Looks good.

**Comment by karlbaumhauer** [comment]
@saemik94 I was not part of any discussion here but it seems wrong, that I see the hover effect on the image while hovering on the link in the teaser bottom. 
In general the UX here seems to be not 100% clear to me. Is this really the expected behavior @coraliefeil? From my point of view (in case we have a link or a button in the teaser bottom), it should only show the pointer cursor and the hover effect if I interact with the image. 
Might be complicated but the whole idea of having a link in the teaser and another link in the teaser bottom (or even 2 of them) seems a wrong use case to me. Or what would be the story behind it?

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 9]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+            </div>
+          </sd-teaser>
+        </a>
+      </div>
+    `;
+  }
+};
+
+/**
+ * ## Clickable teaser with extra links
```
Comment (karlbaumhauer) [inline packages/components/src/components/teaser/teaser.stories.ts:569]: unfortunately this sample needs to be removed. We agreed not to provide a sample with nested links, as we dont want the users to use the teaser like this (UX, a11y - even though you found a workaround).

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 10]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+    `;
+  }
+};
+
+/**
+ * ## Clickable teaser
+ *
+ * If the teaser itself should be clickable and there are no other links inside, then wrap the teaser with an anchor tag.
+ * The button on the bottom of the teaser is not a link, it's just to make the interaction more obvious. According to
+ * the HTML spec, it is not allowed to have an anchor tag inside another anchor.
```
Comment (karlbaumhauer) [inline packages/components/src/components/teaser/teaser.stories.ts:467]:
```suggestion
 * If the teaser itself should be clickable and there are no other links inside, then wrap the teaser with an anchor tag.
 * The button on the bottom of the teaser is **must not be a link**. According to the HTML spec, it is not allowed to have an anchor tag inside another anchor.
```

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 11]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+ *
+ * ```html
+ * <a href="#">
+ *    <sd-teaser class="interactive">
+ *      <sd-button>More</sd-button>
```
Comment (karlbaumhauer) [inline packages/components/src/components/teaser/teaser.stories.ts:472]:
```suggestion
 *      <sd-button>CTA</sd-button>
```

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 12]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+              <span class="meta-info-item">| Author name</span>
+            </div>
+            <h3 slot="headline">Clickable teaser</h3>
+            <div class="flex flex-col gap-5">
+              <p>
+                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
+                dolore magna aliqua.
+              </p>
+              <div class="flex-none">
+                <sd-button variant="primary" inverted>More</sd-button>
```
Comment (karlbaumhauer) [inline packages/components/src/components/teaser/teaser.stories.ts:536]:
```suggestion
                <sd-button variant="cta" inverted>Add to shopping cart</sd-button>
```

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 13]
```diff
@@ -382,3 +382,361 @@ export const Parts = {
...
+              <span class="meta-info-item">| Author name</span>
+            </div>
+            <h3 slot="headline">Clickable teaser</h3>
+            <div class="flex flex-col gap-5">
+              <p>
+                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
+                dolore magna aliqua.
+              </p>
+              <div class="flex-none">
+                <sd-button variant="primary">More</sd-button>
```
Comment (karlbaumhauer) [inline packages/components/src/components/teaser/teaser.stories.ts:558]:
```suggestion
                <sd-button variant="cta" inverted>Add to shopping cart</sd-button>
```

File: packages/components/src/components/teaser/teaser.stories.ts (hunk 10 above)
Comment (karlbaumhauer) [reply packages/components/src/components/teaser/teaser.stories.ts:467]: @solid-design-system/product-team we actually forgot the use case of a button in combination with a clickable teaser. the teaser could be a link to more details of a workshop (link on the teaser) but we need a button to add it to the shopping cart. 

I renamed the button texts and changed the description a bit. I hope this covers all we talked about ealier today. If not, lets chat tomorrow in the daily.

**Review by karlbaumhauer — CHANGES_REQUESTED** [review]

File: packages/components/src/components/teaser/teaser.stories.ts (hunk 9 above)
Comment (saemik94) [reply packages/components/src/components/teaser/teaser.stories.ts:569]: at least I learned something while implementing ;) I think I didn't understand the ticket and the provided .gif a 100% then.

File: packages/components/src/components/teaser/teaser.stories.ts (hunk 10 above)
Comment (saemik94) [reply packages/components/src/components/teaser/teaser.stories.ts:467]: I applied your suggested changes, nevertheless lets have a quick look at it after the daily :)

**Review by karlbaumhauer — APPROVED** [review]

**Comment by coraliefeil** [comment]
Review done, pls reassign when adjusted.

**Comment by coraliefeil** [comment]
Approved

**Pull request merged**